Extensions/probing.hh

```diff
@@ -232,94 +232,87 @@ static double CalculatePseudoEnergy(double data, const std::string &modifier,
   return val;
 }
 
-// END STOLEN FROM RNASTRUCTURE
-static double calculateScore(const Subsequence &Base, const bool isUnpaired,
-                             const std::vector<double> &probingData,
-                             const double clusterPaired,
-                             const double clusterUnpaired,
-                             const bool hasDMSModifier,
-                             const bool hasCMCTModifier,
-                             const bool isCentroidProbNorm) {
-  double score = 0.0;
-  for (unsigned int i = Base.i; i < Base.j && i < probingData.size(); i++) {
-    if (hasDMSModifier && (Base[i] != A_BASE) && (Base[i] != C_BASE)) {
-      continue;
-    } else if (hasCMCTModifier && (Base[i] != U_BASE) && (Base[i] != G_BASE)) {
-      continue;
-    }
-    if (isCentroidProbNorm) {
-      if (isUnpaired) {
-        score += fabs(probingData.at(i) - clusterUnpaired);
-      } else {
-        score += fabs(probingData.at(i) - clusterPaired);
-      }
-    } else {
-      if (isUnpaired) {
-        score += probingData.at(i);
+static void calcBaseScores(double **baseScores,
+                           const Subsequence &Base,
+                           const std::vector<double> &probingData,
+                           const double clusterPaired,
+                           const double clusterUnpaired,
+                           const bool hasDMSModifier,
+                           const bool hasCMCTModifier,
+                           const bool isCentroidProbNorm) {
+  /* precalculate the score from the first base 
+     up to every base position for cheap and fast
+     score calculation in getReactivityScore
+  */
+
+  const unsigned int seqLen = Base.seq->n;
+  const unsigned int lookupSize = seqLen + 1;
+
+  // allocate space and store pointers in baseScores
+  baseScores[0] = new double[lookupSize];  // paired base scores
+  baseScores[1] = new double[lookupSize];  // unpaired base scores
+  baseScores[0][0] = 0.0;
+  baseScores[1][0] = 0.0;
+
+  double pairedScore = 0.0, unpairedScore = 0.0;
+
+
+  for (unsigned int i = 0; i < probingData.size() && i < seqLen; i++) {
```

Comment:
will this also work for co-folding situations, where the probing data file contains a separator to split probing information for the two input nucleotide sequences?

Reply:
I didn't change they way the probing information is parsed, so this should also work in those situations.